packages/stores/src/queries-external/ibc-status/index.ts

```diff
@@ -0,0 +1,49 @@
+import { KVStore } from "@keplr-wallet/common";
+import { HasMapStore } from "@keplr-wallet/stores";
+import { makeObservable } from "mobx";
+import { computedFn } from "mobx-utils";
+import { ObservableQueryExternalBase } from "../base";
+import { IbcStatus } from "./types";
+
+export class ObservableQueryIbcStatus extends ObservableQueryExternalBase<IbcStatus> {
+  constructor(kvStore: KVStore, baseURL: string, _counterPartyChainID: string) {
+    super(kvStore, baseURL, `/ibc/v1/all`);
```

Comment:
It could get space-inefficient to query all statuses and then store them for every osmosis-counterparty pair

Reply:
From dilan:

instead of for each counterPartyChainID , if you want:
from X to osmosis => https://api-osmosis.imperator.co/ibc/v1/source/counterPartyChainID/destinationosmosis-1?minutes_trigger=-1
from osmosis to X => https://api-osmosis.imperator.co/ibc/v1/source/osmosis-1/destinationcounterPartyChainID?minutes_trigger=-1

Reply:
Yeah that makes sense to use the source<>destination query. At first I thought there was a way to use counterPartyChainID as the only key, but would be inefficient to loop through all the pairs. Will we always be transferring between X and Osmosis? I think there are other pairs right? So we would need to pass in both the source and the destination to the `ObservableQueryExternalBase `constructor

Reply:
For our purposes, we should only expect to query for Osmosis and the counterparty. Though, we may need to create a map store for each direction, deposit or withdraws.

Perhaps, you could create another store that wraps 2 map stores for each direction. Then has functions that accept a direction (of type `"deposit" | "withdraw") and a counterparty chain ID. Then it calls the appropriate map store depending on direction.

Reply:
Hmmm ok, for now I made two queries, one for withdrawing and one for depositing, not ideal though as it has some duplicate code